Wikipedia/Code/MWKDataStore.m

```diff
@@ -495,6 +529,53 @@ - (void)iterateOverArticles:(void (^)(MWKArticle*))block {
     }
 }
 
+- (void)startCacheRemoval {
+    dispatch_async(self.cacheRemovalQueue, ^{
+        self.cacheRemovalActive = true;
```

Comment:
What do you think of using an NSOperationQueue as an ivar and then using an NSOperation? Then you could just check if it has any current operations.

This would be instead of using GCD plus a flag to check if cache removal was in progress.

The main thought here is that that it would eliminate the extra variable for tracking state and it would use a higher level API.